fgpyo/sam/__init__.py

```diff
@@ -265,7 +265,8 @@ def _pysam_open(
             assert file_type is not None, "Must specify file_type when writing to standard output"
             path = sys.stdout
         else:
-            file_type = file_type or SamFileType.from_path(path)
+            if file_type is None:
+                file_type = SamFileType.from_path(path)
```

Comment:
I must have a newer MyPy in my virtual environment because it complained about this pattern.